src/dataprocessor/dataprocessor.js

```diff
@@ -0,0 +1,49 @@
+/**
+ * @license Copyright (c) 2003-2016, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+'use strict';
+
+import Config from '../config.js';
+
+/**
+ * Abstract base DataProcessor class. This class should be extended by actual DataProcessor implementations.
+ *
+ * @abstract
+ * @class dataProcessor.DataProcessor
+ */
+export default class DataProcessor {
+	/**
+	 * Creates a new instance of DataProcessor class.
+	 *
+	 *
+	 * @param {Object} config DataProcessor's configuration.
+	 * @constructor
+	 */
+	constructor( config ) {
+		/**
+		 * Holds configurations specific to this DataProcessor instance.
+		 *
+		 * @readonly
+		 * @property {Config}
+		 */
+		this.config = new Config( config );
```

Comment:
1. The `Config` class is not for this. It's for handling the editor config (although it was implemented as a generic tool).
2. In the `DataProcessor` we don't need any config. As you can see – it does not use any. If some child classes will need some options their constructors will override this one and set those options.

In other words – the constructor isn't needed in this class.